components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -452,95 +461,138 @@ private String getOIDCAuthzEndpoint(Map<String, String> authenticatorProperties)
-                }
+        return oAuthResponse.getParam(OIDCAuthenticatorConstants.ID_TOKEN);
+    }
 
-                String authenticatedUserId = getAuthenticatedUserId(context, oAuthResponse, jsonObject);
-                String attributeSeparator = getMultiAttributeSeparator(context, authenticatedUserId);
+    /**
+     * Retrieves or maps the access token according to the flow supported by the authenticator.
+     * Overridden in Google Authenticator for Google one tap.
+     *
+     * @param request       HttpServletRequest.
+     * @param context       AuthenticationContext.
+     * @param oAuthResponse OAuthClientResponse.
+     * @throws AuthenticationFailedException Throws error when access token is not found.
+     */
+    protected void mapAccessToken(HttpServletRequest request, AuthenticationContext context,
+                                  OAuthClientResponse oAuthResponse) throws AuthenticationFailedException {
 
-                jsonObject.entrySet().stream()
-                        .filter(entry -> !ArrayUtils.contains(NON_USER_ATTRIBUTES, entry.getKey()))
-                        .forEach(entry -> buildClaimMappings(claims, entry, attributeSeparator));
-                
-                authenticatedUser = AuthenticatedUser
-                        .createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedUserId);
-            } else {
+        String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
 
-                if (log.isDebugEnabled()) {
-                    log.debug("The IdToken is null");
-                }
-                authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
-                        getAuthenticateUser(context, jsonObject, oAuthResponse));
-            }
+        if (StringUtils.isBlank(accessToken)) {
+            throw new AuthenticationFailedException(ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getCode(),
+                    ErrorMessages.ACCESS_TOKEN_EMPTY_OR_NULL.getMessage());
+        }
+        context.setProperty(OIDCAuthenticatorConstants.ACCESS_TOKEN, accessToken);
+    }
 
-            claims.putAll(getSubjectAttributes(oAuthResponse, authenticatorProperties));
-            authenticatedUser.setUserAttributes(claims);
+    /**
+     * Generates OAuth client and returns the oAuthResponse according to the flow supported by the authenticator.
+     * Overridden in Google Authenticator for Google one tap.
+     *
+     * @param request HttpServletRequest.
+     * @param context AuthenticationContext.
+     * @return OAuthClientResponse.
+     * @throws AuthenticationFailedException throws error when OAuthAuthzResponse validation fails for either error
+     *                                       response or the parameters.
+     */
+    protected OAuthClientResponse generateOauthResponse(HttpServletRequest request, AuthenticationContext context)
```

Comment:
`requestAccessToken` WDYT?

Reply:
This is fixed 17aec3bd27b4fb46de26ee69631363ceabb32f2c